### PR TITLE
Mjberry/update json format

### PR DESCRIPTION
**Pull request by matthewberry**
This PR adopts the revised data format. It's a big one. I'll add lots of comments on the files tab.

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -41,12 +41,12 @@
                 {
                   "type": "initial",
                   "maximumWarning": "500kb",
```
Comment (matthewberry) [inline angular.json:43]: I increased these budgets for the time being. Big things in our build include the data file and some of the scss files.

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -12,7 +12,10 @@ const appRoutes: Routes = [
 
 @NgModule({
   imports: [
-    RouterModule.forRoot(appRoutes)
```
Comment (matthewberry) [inline src/app/app-routing.module.ts:15]: There are a few small changes to simplify deployment to github pages (e.g., https://ncsa.github.io/phylodiver-viz/#/demo). I might come back and clean them up in a later PR.

File: src/app/app.module.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { BrowserModule } from '@angular/platform-browser';
 
 import { AppComponent } from './app.component';
 import { LandingPageComponent } from './components/page/landing-page/landing-page.component';
-import { LegendNodeComponent } from './components/legend-node/legend-node.component';
```
Comment (matthewberry) [inline src/app/app.module.ts:6]: renaming (but I'm not totally happy with my new name)

File: src/app/components/node/node.component.html [hunk 4]
```diff
@@ -1,101 +1,80 @@
-<div class="block" [class.is_parent_block]="!children || children.length === 0">
```
Comment (matthewberry) [inline src/app/components/node/node.component.html:1]: lots of changes to the data model and a few changes to names in this file

File: src/app/components/node/node.component.html [hunk 5]
```diff
@@ -1,101 +1,80 @@
...
             </div>
-            <div class="summary_button" (click)="$event.stopPropagation(); onToggleTable(undefined)">{{(!phylogenyShowTable) ? 'Open in' : 'Hide'}} Table</div>
+            <div class="summary_button" (click)="$event.stopPropagation(); onToggleTable(null)">{{(!showTable) ? 'Open in' : 'Hide'}} Table</div>
         </div>
     </div>
 
-    <div class="block_background" *ngIf="children && children.length > 0">
+    <div class="block_background" *ngIf="displayNode.children.length > 0">
         <svg width="100%" height="100%" viewBox="0 0 100 100" preserveAspectRatio="none">
-            <linearGradient x1="0" y1="0" x2="100%" y2="0" id="gradient">
```
Comment (matthewberry) [inline src/app/components/node/node.component.html:72]: if, for a given instance of this component, we want its .block_background to use its linearGradient, the id here must be unique on the page

File: src/app/components/node/node.component.html [hunk 6]
```diff
@@ -1,101 +1,80 @@
...
+            <linearGradient x1="0" y1="0" x2="100%" y2="0" [id]="gradientId">
                 <stop class="gradient_start" offset="0" />
                 <stop class="gradient_end" offset="100%" />
             </linearGradient>
-            <path id="curve_background" style="stroke-miterlimit:10;" fill="url(#gradient)" d="M0 0 C 0 20, 100 20, 100 0 L 100 100 C 100 80, 0 80, 0 100 L 0 0" vector-effect="non-scaling-stroke"></path>
```
Comment (matthewberry) [inline src/app/components/node/node.component.html:76]: and changed these from ids to classes so we don't have to worry about uniqueness of ids

File: src/app/components/node/node.component.ts [hunk 7]
```diff
@@ -22,102 +23,75 @@ export class NodeComponent implements OnInit, OnChanges, OnDestroy {
...
-    't1_cgc_genes': 0,
-    'genes': 0,
-    'drugs': 0,
-  };
-  phylogenyProportionId: any; // TODO FIXME
-  phylogenyShowTable = false;
-  selectedBlocks: NodeComponent[] = [];
+  gradientId = 'uninitialized';
 
   summaryColumns = [
```
Comment (matthewberry) [inline src/app/components/node/node.component.ts:41]: chatted with Charles about what's possible with the new pipeline data and arrived at this set of columns for now

File: src/app/components/node/node.component.ts [hunk 8]
```diff
@@ -130,102 +104,119 @@ export class NodeComponent implements OnInit, OnChanges, OnDestroy {
...
+          if (geneName && geneName.length > 0) {
+            variantInfo.genesAll.add(geneName);
+            if (snp.cgcGeneInfo) {
+              variantInfo.genesCgc.add(geneName);
+              if (snp.cgcGeneInfo.tier === 1) {
+                variantInfo.mutationsTier1.add(variant);
+              }
+            }
+          }
+          // TODO drugs
```
Comment (matthewberry) [inline src/app/components/node/node.component.ts:140]: this comment didn't seem funny when I wrote it

File: src/app/components/node/node.component.ts [hunk 9]
```diff
@@ -130,102 +104,119 @@ export class NodeComponent implements OnInit, OnChanges, OnDestroy {
...
       '--cell_block_color': startColor,
       '--gradient_start': startColor,
       '--gradient_end': childrenColor
     } as any;
-    //only apply the height for the final tree end
-    if (this.parentTotal) {
-      const hasSelectedBlock = this.selectedBlocks.length > 0;
-      const selectedBlockLevel = hasSelectedBlock ? this.selectedBlocks[0].blockLevel : 0;
-      styles['flex'] = (!this.ignoreScaling && hasSelectedBlock && selectedBlockLevel >= this.blockLevel && !this.isSelected) ? '1' : '' + (this.getProportion() * 1000); //proper value is 100, using 1000 to force this to shrink super small when selected
+    //only apply the height for the final tree end // TODO FIXME not what I'm doing at the moment; see next line
```
Comment (matthewberry) [inline src/app/components/node/node.component.ts:163]: I was having problems getting the block sizes right so I cheated and set them on each block for the time being. at some point we can take another look at it (or maybe it'll be obvious to you, Chad!)

File: src/app/components/single-sample-viewer/single-sample-viewer.component.html [hunk 10]
```diff
@@ -1,33 +1,27 @@
...
+            <phylo-legend-sample *ngFor="let legendSample of legendSamples"
+              [legendSample]="legendSample">
+            ></phylo-legend-sample>
         </div>
     </div>
 
 <!-- todo: change phylo-table ngIf to angular animation so it can grow/shrink before show/hide -->
-    <phylo-table *ngIf="hasSelectedBlock && phylogenyShowTable"></phylo-table>
+    <phylo-table *ngIf="hasSelectedNode && showTable"></phylo-table>
+    <select (change)="selectTree($any($event.target).value)">
```
Comment (matthewberry) [inline src/app/components/single-sample-viewer/single-sample-viewer.component.html:24]: added an ugly tree selector. Charles and Lisa might have thoughts on where, exactly, this belongs and how it should look

File: src/app/components/single-sample-viewer/single-sample-viewer.component.ts [hunk 11]
```diff
@@ -17,43 +17,54 @@ export class SingleSampleViewerComponent implements OnInit, OnDestroy {
...
-    private changeDetectorRef: ChangeDetectorRef
+    private selectionService: SelectionService
   ) { }
 
   ngOnInit(): void {
-    this.subscriptions.push(this.dataService.getPhylogenyData().subscribe(phyloData => {
-      this.cellData = phyloData.cell_data;
+    this.subscriptions.push(this.dataService.getRootDisplayNode().subscribe(root => {
+      this.rootNode = root;
+      console.log(root);
```
Comment (matthewberry) [inline src/app/components/single-sample-viewer/single-sample-viewer.component.ts:37]: I suppose I can get rid of this console.log now

File: src/app/components/table/table.component.html [hunk 12]
```diff
@@ -4,35 +4,35 @@
...
             </tr>
         </thead>
 
         <tbody>
-            <ng-container *ngFor="let data of detailsData">
-                <tr *ngIf="!phylogenySelectedTier || phylogenySelectedTier === data.tier" (click)="$event.stopPropagation(); selectRow(data)">
-                    <td *ngFor="let column of tableColumns; index as i" [ngClass]="'column_' + column.field">
-                        {{data[column.field]}}
+            <ng-container *ngFor="let row of tableRows">
+                <tr *ngIf="!consequence || consequence === row.variant.consequence" (click)="$event.stopPropagation(); selectRow(row)">
```
Comment (matthewberry) [inline src/app/components/table/table.component.html:28]: this table gets big enough to be slow; we can chat about adding a table from one of our favorite component libraries for pagination, etc.

File: src/app/components/table/table.component.ts [hunk 13]
```diff
@@ -11,57 +14,132 @@ import { SelectionService } from 'src/app/services/selection.service';
...
 
   ngOnDestroy(): void {
     this.subscriptions.forEach(sub => sub.unsubscribe());
   }
 
-  deselectTier() {
-    this.selectionService.setPhylogenySelectedTier(false);
+  updateTableColumns(): void {
+    const columns = [
+      { label: 'Consequence', cssSuffix: 'consequence', accessor: (variant: DisplayVariant, clusterId: number) => variant.consequence ?? '' },
```
Comment (matthewberry) [inline src/app/components/table/table.component.ts:76]: chatted with Charles about the new data and revised the list of columns

File: src/app/services/data.service.ts [hunk 14]
```diff
@@ -1,44 +1,250 @@
...
+              proportion: clusterIdAndSampleIdToPrevalance.get(clusterId + '-' + sample.sample_id) || 0
+            }))
+          });
+        });
+        return returnVal;
+      })
+    );
+  }
+
+  getRootDisplayNode(): Observable<DisplayNode|null> {
```
Comment (matthewberry) [inline src/app/services/data.service.ts:118]: still lots I'd like to do to simplify this and the downstream code

Comment (matthewberry) [reply src/app/services/data.service.ts:118]: this method in particular gets pretty ugly

File: src/app/components/single-sample-viewer/single-sample-viewer.component.ts (hunk 11 above)
Comment (cholson) [inline src/app/components/single-sample-viewer/single-sample-viewer.component.ts:37]: debug left in on purpose? no worries either way

**Review by cholson — APPROVED** [review]

**Pull request merged**